datashape/typesets.py

```diff
@@ -90,8 +89,9 @@ def __init__(self):
         self.lookup = self.registry.get
 
     def register_typeset(self, name, typeset):
-        if name in typeset:
-            raise DataShapeTypeError("TypeSet %s already defined" % name)
+        if name in self.registry:
```

Comment:
`name in typeset` is a bug. we're registering the typeset and typesets contain other dshapes, so it doesn't make sense to look for a string contained in the typeset we're registering.